native/packages/react-native-bpk-component-button/src/BpkButton.js

```diff
@@ -64,13 +66,35 @@
    return styleForElement;
  };
 
- const getGradientColors = ({ type, disabled, selected }) => {
-   let gradientColors = styles.gradientColors[type];
+ const getThemingForElement = (elementType, theme, { type, disabled, selected }) => {
+   const themeForElement = {};
+   if (theme && styles.themeMappings[elementType]) {
+     Object.keys(styles.themeMappings[elementType]).forEach((key) => {
+       const values = styles.themeMappings[elementType][key];
+
+       if (disabled && theme[values.disabled]) {
+         themeForElement[key] = theme[values.disabled];
+       } else if (selected && theme[values.selected]) {
+         themeForElement[key] = theme[values.selected];
+       } else if (values[type]) {
+         themeForElement[key] = theme[values[type]];
+       }
+     });
+   }
+   return themeForElement;
+ };
```

Comment:
Here we iterate over the mappings (set in `BpkButton-styles.js`) and apply them to the component, based on disabled, selected status etc.